src/components/ClientDialog.tsx

```diff
@@ -0,0 +1,207 @@
+/**
+* ClientDialog.js
+*
+* @2023 Digital Aid Seattle
+*/
+
+import { Button } from "primereact/button";
+import { Dialog } from "primereact/dialog";
+import { InputText } from "primereact/inputtext";
+import { RadioButton } from 'primereact/radiobutton';
+
+import React, { useEffect, useState, useRef} from "react";
+import { TicketType, clientService } from "../services/ClientService";
+import { AnimalType, ClientType, RequestType } from "../types";
+
+const defaultClient: ClientType = {
+  id: null,
+  first_name: '',
+  last_name: '',
+  email: '',
+  phone: '',
+}
+
+const defaultAnimal: AnimalType = {
+  id: null,
+  name: '',
+  species: '',
+  client_id: null,
+}
+
+const defaultRequest: RequestType = {
+  id: null,
+  client_id: null,
+  animal_id: null,
+  service_category: '',
+  source: '',
+  staff_id: null,
+}
+
+const ClientDialog = (props) => {
+
+  const [clientDialog, setClientDialog] = useState(false);
+  const [source, setSource] = useState(TicketType.email);
+  const [request, setRequest] = useState<RequestType>(defaultRequest);
+  const [client, setClient] = useState<ClientType>(defaultClient);
+  const [animal, setAnimal] = useState<AnimalType>(defaultAnimal);
+
+  useEffect(() => {
+    setClientDialog(props.visible)
+  }, [props])
+  
+  const hideClientDialog = () => {
+    props.onClose(undefined);
+    setClient(defaultClient);
+  };
+  
+  const saveClientDialog = () => {
+    // type handle separate to support RadioButton
+
+    setRequest(prevRequest => ({...prevRequest, source: source}));
+    clientService.newRequest(request, client, animal)
+      .then(requestResponse => props.onClose(requestResponse))
```

Comment:
Can saveClientDialog be an async function to have consistent Promise usage syntax with line 82? Haven't looked at other files so not sure which style is more common.

Reply:
Currently a mix through repo, with .then syntax more common. We can have a team discussion if we want to enforce one v the other.

Reply:
Use what is appropriate when appropriate, remember **await** blocks, **.then** does not.